### ISSUE TITLE
bug: unexpected dedup during decorrelation

### ISSUE THREAD

**Issue by xiangjinwu** [singularity-data/risingwave#2768]
**Describe the bug**
Follow the repro steps below, and the actual output is:
```
 a1 | a2 
----+----
  2 |  3
(1 row)
```

**To Reproduce**
```
create table a (a1 int, a2 int);
insert into a values (2, 3), (2, 3);

create table b (b1 int, b2 int);
insert into b values (2, 2);

select * from (select * from a) as a where a2 > (select min(b2) from b where a1 = b1);
```

**Expected behavior**
```
 a1 | a2 
----+----
  2 |  3
  2 |  3
(2 rows)
```

**Additional context**
Similar queries are able to give the correct output:
```
select * from a where a2 > (select min(b2) from b where a1 = b1);  -- without from-subquery
select * from (select * from a) as a where a2 > (select min(b2) from b);  -- without correlation
```

https://github.com/singularity-data/risingwave/blob/ea786a919f3fff628dd317ea53cf570e991eb5b6/src/frontend/src/optimizer/rule/unnest_agg_for_loj.rs#L74-L75

During decorrelation, SimpleAgg is transformed into GroupAgg by using all columns on the left/outer side as groups. This is only correct when they are unique.

The alternative query without from-subquery works because `_row_id` is scanned implicitly. But in the buggy query with from-query, `_row_id` is dropped inside and not exposed to outer query.

Maybe we need to include pk_derive or find a lighter-weight alternative? @likg227 @st1page @fuyufjh

**Comment by fuyufjh** [comment]
For more details, this is the plan of the incorrect query (Exchange and Project is omitted)

```
explain select * from (select * from a) as a where a2 > (select min(b2) from b where a1 = b1);
                           QUERY PLAN
-----------------------------------------------------------------
     BatchFilter { predicate: ($1 > $2) }
       BatchHashAgg { group_keys: [$0, $1], aggs: [min($2)] }  <---- PROBLEM IS HERE
           BatchHashJoin { type: LeftOuter, predicate: $0 = $3 }
               BatchScan { table: a, columns: [a1, a2] }
             BatchProject { exprs: [$1, $0] }
                 BatchScan { table: b, columns: [b1, b2] }
```

The problem is caused by the aggregation `group_keys: [$0, $1]`

**Comment by likg227** [comment]
I'm exploring if there is a way to flatten the subquery in FROM clause........

**Comment by likg227** [comment]
This issue get solved in #2880
